torch/_dynamo/variables/nn_module.py

```diff
@@ -388,6 +399,120 @@ def generic_call_method_helper(name):
             # Dynamo inlines `__call__`, includes hooks.
             return self.call_function(tx, args, kwargs)
         elif name == "forward":
+            if config.use_single_step_graph:
+                import torch.utils._pytree as pytree
+                from .builtin import BuiltinVariable
+                from .inline_helper import (
+                    create_functional_call,
+                    decompose_and_inline_function_with_makefx,
+                )
+
+                start_node_code = len(tx.output.graph.nodes)
+
+                # For single step graph we want to use make_fx to extract the fx graph
+                # representing the fwd and inline that fx graph.
+                # Dynamo needs to see all inputs and output of each torch function
+                # in case they are saved for the backward.
+                mod = tx.output.get_submodule(self.module_key)
+                fn = mod.forward.__func__
+                # TODO(JackCaoG): considering using source and regualr builder instead of
+                # SourcelessBuilder.
+                fn_source = AttrSource(self.source, "__call__")
+                fn_source = AttrSource(fn_source, "__func__")
+
```

Comment:
@ezyang I am not sure if I should correctly handle the source of the mod after `make_fx`. I am currently just use a `SourcelessBuilder.`

Reply:
Sourceful things can only come from Dynamo. So I think I would agree they cannot come from the trace. LGTM.